### DISCUSSION TITLE
Combine multiple toolbars

### DISCUSSION THREAD

**Discussion by Steven-AA** [chrisgurney/obsidian-note-toolbar#32] [Ideas]
It would be more effective if multiple toolbars were automatically combined when multiple conditions are satisfied, rather than being overridden.

For instance:

- I have a basic toolbar containing tools 1, 2, and 3. I’ve set the folder mapping for the root directory (“/”).
- Next, I create a toolbar named “A” with tools 4 and 5. I’ve set the folder mapping specifically for the “/A” directory.
- Additionally, I’ve set up a toolbar called “Excalidraw” with tools 6 and 7. For notes related to Excalidraw, I will add “notetoolbar:Excalidraw”.

Expected behavior:
- Excalidraw related note outside "/A" ：1,2,3,6,7
- Excalidraw related note in "/A" :  1,2,3,4,5,6,7
- Normal note in "/A" : 1,2,3,4,5

**Comment by Steven-AA** [comment]
BTW, set folder mapping with "/" is not working for me.

**Comment by FelipeRearden** [comment]
> BTW, set folder mapping with "/" is not working for me.

More info here #27

**Reply by chrisgurney** [reply]
Thank you @FelipeRearden!

**Comment by chrisgurney** [comment]
@Steven-AA Interesting idea!

I could see this as being optional, for sure, so as not to generate any unpredictable toolbars. 

Perhaps each toolbar gets an `Append` flag which, if `true`, would be permission to merge child toolbars with any parents.

**Reply by FeralFlora** [reply]
> Perhaps each toolbar gets an Append flag which, if true, would be permission to merge child toolbars with any parents.

I agree with this idea. See #77

**Comment by Steven-AA** [comment]
It seems that others are also seeking similar enhancements. Are there any plans for this feature?

**Reply by chrisgurney** [reply]
@Steven-AA Yes, this is planned. I've been delayed with other life priorities, but once I get the current beta shipped, it should be next on my list.

**Comment by chrisgurney** [comment]
@Steven-AA @FeralFlora @FelipeRearden You can now (manually) combine toolbars using `Item Groups`. See the [beta release notes](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.10.0-beta-17).

I'd love some feedback on this feature!

Here's the [full beta announcement](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/114) with everything available.
